pkg/scheduler/schedule_one.go

```diff
@@ -66,12 +66,19 @@ const (
 // scheduleOne does the entire scheduling workflow for a single pod. It is serialized on the scheduling algorithm's host fitting.
 func (sched *Scheduler) scheduleOne(ctx context.Context) {
 	logger := klog.FromContext(ctx)
-	podInfo := sched.NextPod()
+	podInfo, err := sched.NextPod()
+	if err != nil {
+		logger.Error(err, "Error while retrieving next pod from scheduling queue")
+		return
+	}
 	// pod could be nil when schedulerQueue is closed
 	if podInfo == nil || podInfo.Pod == nil {
 		return
 	}
+
 	pod := podInfo.Pod
+	logger.V(4).Info("About to try and schedule pod", "pod", klog.KObj(pod))
+
```

Comment:
why not swallow the metric logic which used to locate in MakeNextPodFunc():

https://github.com/kubernetes/kubernetes/blob/ce7fd466aa0c4b7ef000981436a9b20af67a4185/pkg/scheduler/internal/queue/scheduling_queue.go#L1228-L1233